### ISSUE TITLE
feature request: support `compat` with more subtle callout format

### ISSUE THREAD

**Issue by vtjnash** [JuliaDocs/Documenter.jl#2315]
When adding `!!! compat "Julia 1.6"` blocks, these render quite boldly in the docs, while it seems like they could be a more subtle callout in some variation of light gray, smaller, and/or italics:
![image](https://github.com/JuliaDocs/Documenter.jl/assets/330950/d66f0411-0305-44bb-9ed4-1bce11bc53b5)

Compare to say sphinx, where it is rendered with a more subtle italics:
![image](https://github.com/JuliaDocs/Documenter.jl/assets/330950/eb8124ad-b95a-4bd4-9e60-3c6998c14165)

Refs: https://github.com/JuliaLang/julia/pull/46717/files

**Comment by mortenpi** [comment]
Yeah, I think we can de-emphasize the compats a bit. I don't think we want to de-style them completely, like in sphinx -- they're still admonitions, and they should keep looking like ones. Happy to take suggestions for new colors if someone wants to put together a few examples. 

Although, the comment in https://github.com/JuliaLang/julia/pull/46717#issuecomment-1243972433

> Would it be possible to change the rendering so that it is just a tiny badge? 

made me think that maybe we should actually give them more special styling (maybe expand on click etc). But again, would need someone to prototype some ideas as to what they should look like.

Just noting for future reference --- the color itself is set in these places I think:

https://github.com/JuliaDocs/Documenter.jl/blob/5dafb6488f90d173ca4fcfeead0396332bdc6de6/assets/html/scss/documenter/_variables.scss#L18
https://github.com/JuliaDocs/Documenter.jl/blob/5dafb6488f90d173ca4fcfeead0396332bdc6de6/assets/html/scss/documenter-light.scss#L8
https://github.com/JuliaDocs/Documenter.jl/blob/5dafb6488f90d173ca4fcfeead0396332bdc6de6/assets/html/scss/documenter-dark.scss#L33
